### 1. 데이터 찾기 - 의도가 중요

In [2]:
import pandas as pd

df = pd.read_csv("../data/CCTV_Seoul.csv", encoding="cp949")
df.columns

Index(['개방자치단체코드', '관리번호', '관리기관명', '소재지도로명주소', '소재지지번주소', '설치목적구분', '카메라대수',
       '카메라화소수', '촬영방면정보', '보관일수', '설치연월', '관리기관전화번호', 'WGS84위도', 'WGS84경도',
       '데이터기준일자', '데이터갱신구분', '데이터갱신시점', '최종수정시점'],
      dtype='str')

In [3]:
df.head() # 관리기관명으로 groupby한 뒤 갯수 확인하고, 인구수 데이터와 범죄율 데이터 가져와보자!

,개방자치단체코드,관리번호,관리기관명,소재지도로명주소,소재지지번주소,설치목적구분,카메라대수,카메라화소수,촬영방면정보,보관일수,설치연월,관리기관전화번호,WGS84위도,WGS84경도,데이터기준일자,데이터갱신구분,데이터갱신시점,최종수정시점
0,3000000,202630000000800463,서울특별시 종로구청,서울특별시 종로구 북촌로 134-1 (삼청공원),NaN,생활방범,2,200.0,360도 전방면,30.0,202005.0,02-2148-3033,37.58785,126.9843,2026-05-18,NaN,2026-05-19 22:58:23,2026-05-18 15:24:18
1,3000000,202630000000800464,서울특별시 종로구청,서울특별시 종로구 북촌로 134-1 (삼청공원),NaN,생활방범,2,200.0,360도 전방면,30.0,202005.0,02-2148-3033,37.58774,126.9839,2026-05-18,NaN,2026-05-19 22:58:23,2026-05-18 15:24:18
2,3000000,202630000000800465,서울특별시 종로구청,서울특별시 종로구 효자로 29 (마을마당공원),NaN,생활방범,3,200.0,360도 전방면,30.0,202005.0,02-2148-3033,37.57865,126.9736,2026-05-18,NaN,2026-05-19 22:58:23,2026-05-18 15:24:18
3,3000000,202630000000800991,서울특별시 종로구청,서울특별시 종로구 숭인동2길 30,NaN,생활방범,3,200.0,360도 전방면,30.0,201312.0,02-2148-3033,37.57814,127.0202,2026-05-18,NaN,2026-05-19 22:58:23,2026-05-18 15:24:22
4,3000000,202630000000800993,서울특별시 종로구청,서울특별시 종로구 난계로29길 49,NaN,생활방범,3,200.0,360도 전방면,30.0,201312.0,02-2148-3033,37.57380,127.0204,2026-05-18,NaN,2026-05-19 22:58:23,2026-05-18 15:24:22


### 2. 데이터 정제 및 전처리

In [137]:
columns = ["관리기관명", "설치목적구분", "카메라대수"]
clean_df = df[columns]
clean_df.groupby("설치목적구분").count()

,관리기관명,카메라대수
설치목적구분,,
교통단속,6008,6008
교통정보수집,17,17
기타,1388,1388
다목적,226,226
생활방범,44639,44639
시설물관리,1213,1213
쓰레기단속,525,525
어린이보호,4107,4107
재난재해,1501,1501


In [5]:
clean_df.isna().sum()

관리기관명     0
설치목적구분    0
카메라대수     0
dtype: int64

In [103]:
camera_total = clean_df.groupby("관리기관명")[["카메라대수"]].sum()
camera_total = camera_total.reset_index()
camera_total

,관리기관명,카메라대수
0,금천구청,3169
1,서울시 서초구청,6397
2,서울특별시 강남구청,8600
3,서울특별시 강동구청,4237
4,서울특별시 강북구청,4596
5,서울특별시 강서구청,4735
6,서울특별시 관악구청,6789
7,서울특별시 광진구청,1304
8,서울특별시 구로구청,5886
9,서울특별시 노원구청,4240


In [8]:
df2 = pd.read_csv("../data/population.csv")
df2

,동별(1),동별(2),동별(3),2026. 07,2026. 07.1,2026. 07.2,2026. 07.3,2026. 07.4,2026. 07.5,2026. 07.6,2026. 07.7,2026. 07.8,2026. 07.9
0,동별(1),동별(2),동별(3),세대 (세대),합계 (명),합계 (명),합계 (명),한국인 (명),한국인 (명),한국인 (명),외국인 (명),외국인 (명),외국인 (명)
1,동별(1),동별(2),동별(3),소계,소계,남자,여자,소계,남자,여자,소계,남자,여자
2,합계,소계,소계,4522718,9552649,4581706,4970943,9284263,4468307,4815956,268386,113399,154987
3,합계,종로구,소계,72459,147623,70115,77508,136139,65198,70941,11484,4917,6567
4,합계,종로구,청운효자동,4830,10745,4869,5876,10635,4813,5822,110,56,54
...,...,...,...,...,...,...,...,...,...,...,...,...,...
450,합계,강동구,성내2동,13916,24470,11799,12671,23839,11497,12342,631,302,329
451,합계,강동구,성내3동,10665,21867,10554,11313,21509,10416,11093,358,138,220
452,합계,강동구,길동,24205,47908,23108,24800,47421,22947,24474,487,161,326
453,합계,강동구,둔촌1동,11983,33147,16264,16883,33033,16225,16808,114,39,75


In [40]:
mask = (df2["동별(3)"] == "소계") 

gu_total = df2[mask]
gu = gu_total['동별(2)']
gu_total

,동별(1),동별(2),동별(3),2026. 07,2026. 07.1,2026. 07.2,2026. 07.3,2026. 07.4,2026. 07.5,2026. 07.6,2026. 07.7,2026. 07.8,2026. 07.9
2,합계,소계,소계,4522718,9552649,4581706,4970943,9284263,4468307,4815956,268386,113399,154987
3,합계,종로구,소계,72459,147623,70115,77508,136139,65198,70941,11484,4917,6567
21,합계,중구,소계,63854,127602,61067,66535,117493,56335,61158,10109,4732,5377
37,합계,용산구,소계,101944,211750,100963,110787,199575,94772,104803,12175,6191,5984
54,합계,성동구,소계,133729,282264,135636,146628,274346,132283,142063,7918,3353,4565
72,합계,광진구,소계,172185,348371,165187,183184,330802,158095,172707,17569,7092,10477
88,합계,동대문구,소계,180987,373644,179269,194375,353070,171884,181186,20574,7385,13189
104,합계,중랑구,소계,190382,383339,186444,196895,376282,183450,192832,7057,2994,4063
121,합계,성북구,소계,199650,435815,206307,229508,421289,200918,220371,14526,5389,9137
142,합계,강북구,소계,142731,285130,135907,149223,277695,133427,144268,7435,2480,4955


In [124]:
mask2 = (df2.iloc[1] == "소계")
clean_gu_total = gu_total.loc[:, mask2]
clean_gu_total.columns = ["세대 총합", "인구 총합", "한국인 총합", "외국인 총합"]
clean_gu_total['지역'] = gu
population_total = clean_gu_total.copy()
population_total = population_total.reset_index(drop=True)
population_total = population_total.drop(0).reset_index(drop=True)
columns = list(population_total.columns)

columns.remove("지역")
columns.remove("세대 총합")
columns.insert(0, "지역")
population_total = population_total[columns]
population_total

,지역,인구 총합,한국인 총합,외국인 총합
0,종로구,147623,136139,11484
1,중구,127602,117493,10109
2,용산구,211750,199575,12175
3,성동구,282264,274346,7918
4,광진구,348371,330802,17569
5,동대문구,373644,353070,20574
6,중랑구,383339,376282,7057
7,성북구,435815,421289,14526
8,강북구,285130,277695,7435
9,도봉구,301279,298067,3212


In [74]:
df3 = pd.read_csv("../data/crime_fixed.csv", encoding="utf-8-sig")
df3

,구분,죄종,발생검거,건수
0,중부,살인,발생,3
1,중부,살인,검거,2
2,중부,강도,발생,1
3,중부,강도,검거,1
4,중부,"강간,추행",발생,108
...,...,...,...,...
305,수서,"강간,추행",검거,162
306,수서,절도,발생,974
307,수서,절도,검거,515
308,수서,폭력,발생,1234


In [102]:
mask = df3["발생검거"] == "발생"
clean_df3 = df3[mask]
# clean_df3

crime_total = clean_df3.groupby("구분")[["건수"]].sum()
crime_total = crime_total.reset_index()
crime_total

,구분,건수
0,강남,3707
1,강동,3350
2,강북,2289
3,강서,4008
4,관악,4485
5,광진,2870
6,구로,3378
7,금천,2017
8,남대문,1149
9,노원,3101


camera_total, population_total, crime_total 모였다!

In [133]:
camera_total["관리기관명"] = camera_total["관리기관명"].str.replace(
      r"^(서울특별시|서울시)\s*",
      "",
      regex=True
  )

camera_total["관리기관명"] = camera_total["관리기관명"].str.replace(
      r"^(서울특별시|서울시)\s*|\s*청$",
      "",
      regex=True
  )

camera_total = camera_total.rename(
      columns={"관리기관명": "지역"}
  )

camera_total

,지역,카메라대수
0,강남구,8600
1,강동구,4237
2,강북구,4596
3,강서구,4735
4,관악구,6789
5,광진구,1304
6,구로구,5886
7,금천구,3169
8,노원구,4240
9,도봉구,3112


In [138]:
# population_total = population_total.sort_values("지역").reset_index(drop=True)
# camera_total = camera_total.sort_values("관리기관명").reset_index(drop=True)
crime_total

,구분,건수
0,강남,3707
1,강동,3350
2,강북,2289
3,강서,4008
4,관악,4485
5,광진,2870
6,구로,3378
7,금천,2017
8,남대문,1149
9,노원,3101


In [139]:
station_to_gu = {
      "강남": "강남구",
      "강동": "강동구",
      "강북": "강북구",
      "강서": "강서구",
      "관악": "관악구",
      "광진": "광진구",
      "구로": "구로구",
      "금천": "금천구",
      "남대문": "중구",
      "노원": "노원구",
      "도봉": "도봉구",
      "동대문": "동대문구",
      "동작": "동작구",
      "마포": "마포구",
      "방배": "서초구",
      "서대문": "서대문구",
      "서부": "은평구",
      "서초": "서초구",
      "성동": "성동구",
      "성북": "성북구",
      "송파": "송파구",
      "수서": "강남구",
      "양천": "양천구",
      "영등포": "영등포구",
      "용산": "용산구",
      "은평": "은평구",
      "종로": "종로구",
      "종암": "성북구",
      "중랑": "중랑구",
      "중부": "중구",
      "혜화": "종로구"
  }

In [140]:
crime_with_gu = crime_total.copy()

crime_with_gu["지역"] = crime_with_gu["구분"].map(station_to_gu)

crime_with_gu

,구분,건수,지역
0,강남,3707,강남구
1,강동,3350,강동구
2,강북,2289,강북구
3,강서,4008,강서구
4,관악,4485,관악구
5,광진,2870,광진구
6,구로,3378,구로구
7,금천,2017,금천구
8,남대문,1149,중구
9,노원,3101,노원구


In [141]:
crime_gu_total = crime_with_gu.groupby(
      "지역",
      as_index=False
  )["건수"].sum()

crime_gu_total

,지역,건수
0,강남구,6107
1,강동구,3350
2,강북구,2289
3,강서구,4008
4,관악구,4485
5,광진구,2870
6,구로구,3378
7,금천구,2017
8,노원구,3101
9,도봉구,1741


### 3. 병합 및 집계

In [134]:
total_df = pd.merge(
      population_total,
      camera_total,
      on="지역",
      how="left"
  )
total_df

,지역,인구 총합,한국인 총합,외국인 총합,카메라대수
0,강남구,558809,552631,6178,8600
1,강동구,503495,498701,4794,4237
2,강북구,285130,277695,7435,4596
3,강서구,554922,548268,6654,4735
4,관악구,497177,477503,19674,6789
5,광진구,348371,330802,17569,1304
6,구로구,404394,382531,21863,5886
7,금천구,235173,222421,12752,3169
8,노원구,485931,480549,5382,4240
9,도봉구,301279,298067,3212,3112


In [142]:
final_df = pd.merge(
      total_df,
      crime_gu_total,
      on="지역",
      how="left"
  )
final_df

,지역,인구 총합,한국인 총합,외국인 총합,카메라대수,건수
0,강남구,558809,552631,6178,8600,6107
1,강동구,503495,498701,4794,4237,3350
2,강북구,285130,277695,7435,4596,2289
3,강서구,554922,548268,6654,4735,4008
4,관악구,497177,477503,19674,6789,4485
5,광진구,348371,330802,17569,1304,2870
6,구로구,404394,382531,21863,5886,3378
7,금천구,235173,222421,12752,3169,2017
8,노원구,485931,480549,5382,4240,3101
9,도봉구,301279,298067,3212,3112,1741


### 4. 인사이트 정립

In [148]:
analysis_df = final_df.copy()
analysis_df.dtypes

지역          str
인구 총합       str
한국인 총합      str
외국인 총합      str
카메라대수     int64
건수        int64
dtype: object

In [150]:
columns = ["인구 총합", "한국인 총합", "외국인 총합"]

analysis_df[columns] = analysis_df[columns].apply(pd.to_numeric)

In [151]:
analysis_df['인구 1만명당 카메라 수'] = analysis_df['카메라대수'] / analysis_df['인구 총합'] * 10000
analysis_df['인구 1만명당 범죄 건수'] = analysis_df['건수'] / analysis_df['인구 총합'] * 10000
analysis_df

,지역,인구 총합,한국인 총합,외국인 총합,카메라대수,건수,인구 1만명당 카메라 수,인구 1만명당 범죄 건수
0,강남구,558809,552631,6178,8600,6107,153.898738,109.285999
1,강동구,503495,498701,4794,4237,3350,84.151779,66.534921
2,강북구,285130,277695,7435,4596,2289,161.189633,80.279171
3,강서구,554922,548268,6654,4735,4008,85.327307,72.226367
4,관악구,497177,477503,19674,6789,4485,136.550967,90.209322
5,광진구,348371,330802,17569,1304,2870,37.431359,82.383436
6,구로구,404394,382531,21863,5886,3378,145.551121,83.532397
7,금천구,235173,222421,12752,3169,2017,134.751864,85.766648
8,노원구,485931,480549,5382,4240,3101,87.255186,63.815645
9,도봉구,301279,298067,3212,3112,1741,103.292961,57.786968


In [152]:
# 1. 인구 1만 명당 카메라 수 순위
camera_rank = analysis_df.sort_values(
      "인구 1만명당 카메라 수",
      ascending=False
  )[["지역", "인구 1만명당 카메라 수"]]

camera_rank

,지역,인구 1만명당 카메라 수
18,양천구,365.026008
20,용산구,303.990555
23,중구,275.779376
22,종로구,210.536299
2,강북구,161.189633
0,강남구,153.898738
14,서초구,152.484518
6,구로구,145.551121
24,중랑구,144.754382
19,영등포구,140.365434


In [153]:
# 2. 인구 1만 명당 범죄 건수 순위
crime_rank = analysis_df.sort_values(
      "인구 1만명당 범죄 건수",
      ascending=False
  )[["지역", "인구 1만명당 범죄 건수"]]

crime_rank

,지역,인구 1만명당 범죄 건수
23,중구,231.579442
22,종로구,187.301437
20,용산구,156.883117
0,강남구,109.285999
19,영등포구,106.056711
12,마포구,105.937799
14,서초구,90.294099
4,관악구,90.209322
10,동대문구,86.071234
7,금천구,85.766648


In [154]:
# 3. 두 지표의 상관계수
correlation = analysis_df[
      ["인구 1만명당 카메라 수", "인구 1만명당 범죄 건수"]
  ].corr().iloc[0, 1]

correlation

np.float64(0.5548640468527395)

In [155]:
camera_mean = analysis_df["인구 1만명당 카메라 수"].mean()
crime_mean = analysis_df["인구 1만명당 범죄 건수"].mean()

# 카메라도 평균보다 많고 범죄도 평균보다 많은 구
high_camera_high_crime = analysis_df[
      (analysis_df["인구 1만명당 카메라 수"] >= camera_mean) &
      (analysis_df["인구 1만명당 범죄 건수"] >= crime_mean)
  ]

high_camera_high_crime

,지역,인구 총합,한국인 총합,외국인 총합,카메라대수,건수,인구 1만명당 카메라 수,인구 1만명당 범죄 건수
0,강남구,558809,552631,6178,8600,6107,153.898738,109.285999
20,용산구,211750,199575,12175,6437,3322,303.990555,156.883117
22,종로구,147623,136139,11484,3108,2765,210.536299,187.301437
23,중구,127602,117493,10109,3519,2955,275.779376,231.579442


In [156]:
# 카메라는 평균보다 적은데 범죄는 평균보다 많은 구
low_camera_high_crime = analysis_df[
      (analysis_df["인구 1만명당 카메라 수"] < camera_mean) &
      (analysis_df["인구 1만명당 범죄 건수"] >= crime_mean)
]

low_camera_high_crime

,지역,인구 총합,한국인 총합,외국인 총합,카메라대수,건수,인구 1만명당 카메라 수,인구 1만명당 범죄 건수
12,마포구,367291,356154,11137,4088,3891,111.301393,105.937799
19,영등포구,392903,371691,21212,5515,4167,140.365434,106.056711


### 최종 결론(인사이트)
- 인구 1만명당 카메라 수와 인구 1만명당 범죄 건수의 상관관계는 그렇게 높지도 않고 낮지도 않다

    - 서울시 자치구별 분석 결과, 인구 1만 명당 카메라 수와 범죄 건수 사이의 상관계수는 약 0.555로 나타났다. 두 지표는 중간 정도의 양의 상관관계를 보였지만, 이를 카메라가 범죄 발생을 증가시킨다는 인과관계로 해석할 수는 없다.

- 강남, 용산, 종로, 중구는 카메라도 많고 범죄율도 높다
    - 인과관계 정확히 설명하기 어려움. 상권·관광지·업무지구처럼 유동 인구가 많은 지역일 가능성이 큽니다. 범죄 위험이 높거나 신고·관리 수요가 많아서 카메라도 상대적으로 많이 설치됐다고 해석할 수 있다.

- 마포, 영등포구는 카메라는 적은데 범죄율은 높다.
    - 유동 인구나 상업 활동에 비해 CCTV 설치가 부족할 가능성을 살펴볼 수 있다. 조심해야 할지도...?

#### 약간의 개선사항

1. 카메라 데이터에 설치목적 보면 방범만 있는 게 아니라 교통, 쓰레기단속, 기타 등 다양한 항목이 있던데, 특정 설치구분(방범 등)만 뽑으면 조금 더 정확한 분석이 가능할 것 같다.

2. 경찰서 기준 범죄 데이터를 행정구 기준으로 변환해 사용했으므로 결과 해석에는 한계가 있다.